### DISCUSSION TITLE
Read saved values

### DISCUSSION THREAD

**Discussion by eduardoarantes** [Hieromon/AutoConnect#583] [How can I?] [answered]
I am using the Elements example.

The values are stored as JSON. When I read them, I need only the value the user inputted.

I don't seem to find an easy way.

Tks

**Comment by Hieromon** [comment, marked as answer]
Use [ArduinJson API](https://arduinojson.org/v6/api/). The basic steps are:

1. Open the JSON file your sketch saved.
2. Use `deserializeJson` of ArduinoJson function with the `Stream&` argument.
3. Check `DeserializationError`.
4. Close the JSON file.
5. Read value from JsonDocument given the key. AutoConnectElements keys are [documented](https://hieromon.github.io/AutoConnect/acjson.html).

See the [ArduinoJson tutorial](https://arduinojson.org/v6/doc/deserialization/) for details. It's straightforward. Also https://arduinojson.org/v6/example/parser/
